### ISSUE TITLE
🐛 Bug Report: Logout from backstage redirecting to backstage home screen

### ISSUE THREAD

**Issue by vinaykv23** [backstage/backstage#26440]
### 📜 Description

Backstage  "version": "1.28.0"
login to backstage using okta authentication
goto settings -> profile -> click on signout

![image](https://github.com/user-attachments/assets/5f1ee379-9016-4d06-a94e-d29617a3ac8d)

### 👍 Expected behavior

it should logout to okta logout page

### 👎 Actual Behavior with Screenshots

its actually redirecting to backstage home screen

![image](https://github.com/user-attachments/assets/f90e2562-e7c9-4bc9-b780-73d191995e49)

### 👟 Reproduction steps

Backstage  "version": "1.28.0"
login to backstage using okta authentication
goto settings -> profile -> click on signout

it should logout to okta logout page

its actually redirecting to backstage home screen

### 📃 Provide the context for the Bug.

these are some references to the issue
https://github.com/backstage/backstage/issues/25798
https://github.com/backstage/backstage/issues/12016

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by Rugvip** [comment]
We discussed this in the maintainer team the other day. As far as I know the issue here is that the auto option for the built-in sign-in page should not apply directly after logging out. Happy for any help we can get with getting this fixed

**Comment by mahendra1290** [comment]
@Rugvip I can pick this up

**Comment by vinaykv23** [comment]
hi @mahendra1290 
Please let me know if any updates

**Comment by mahendra1290** [comment]
> hi @mahendra1290 Please let me know if any updates

Just started to look into this, was acquired in some other work.

**Comment by mahendra1290** [comment]
@Rugvip I have seen the code on sign-out we call this method, and this path can be either ```app.baseUrl``` or `/`, no custom URL present.

```typescript
// backstage/packages/core-app-api/src/apis/implementations/IdentityApi/AppIdentityProxy.ts
  async signOut(): Promise<void> {
    await this.waitForTarget.then(target => target.signOut());
    await this.#cookieAuthSignOut?.();

    window.location.href = this.signOutTargetUrl;
  }
```

@vinaykv23 on sign out it signs you out of backstage right?, you want it to redirect to okta logout page? where you can logout and then redirected back to backstage is that what you think is correct flow? Just want to understand

**Comment by vinaykv23** [comment]
hi @mahendra1290  
we have upgraded Backstage "version": "1.33.4"
still the logout issue is not resolved
do we need to upgrade any other packsges

**Comment by benjdlambert** [comment]
#26443 is the fix here

**Comment by mcintoac-aws** [comment]
Any updates on this?

I have a related issue and am wondering if it should be a part of this discussion, or a separate ticket.

On top of the auto functionality not redirecting to the sign-in package again on sign-out, for custom OAuth 2 providers, we require the ability to customize the sign-out functionality. This is necessary for Amazon Cognito, which via the hosted-ui, creates an auth-2 refresh cookie. This cookie requires hitting a Cognito logout API to invalidate. Without the ability to customize the signout functionality for our custom provider, we can not invalidate this cookie, and even without the automatic redirect, clicking the "sign-in" button would automatically sign-in the last user rather than allow a different user sign-in.

There is a related issue specifically for the oauth2 proxy provider, but not custom providers. @Rugvip this was an issue you opened:

https://github.com/backstage/backstage/issues/12016

**Comment by vinzscam** [comment]
yep, I think #12016 is the right place to continue the discussion of your issue
